app/views/posts/_post.html.erb

```diff
@@ -6,6 +6,10 @@
     <%= link_to 'Unlike post', blog_post_like_path(post.blog, post), method: :delete %>
   <% end %>
   </br>
+  <% current_user.blogs.each_with_index do |blog, index| %>
+    <%= link_to "Re-post on my #{(index+1).ordinalize} blog", {controller: "posts", action: "repost", post_id: post.id, blog_id: blog.id}, method: :post %>
```

Comment:
I don't think this is ideal, I couldn't stop it sending the id of the blog being viewed. The blog_id param is of the blog the current_user wants to re-post on to.  Looking forward to feedback here!

Reply:
1. I might have approached it something like this:

```
  resources :blogs do
    resources :posts do
       member do
         post 'repost'
       end
    end
  end
```

Then the path helper would look something like 

```
repost_blog_post(post.blog, post, new_blog_id: blog.id)
```

Then the `new_blog_id` param is the blog you're reposting it to. 

But yeah, what you've done is just fine!

2. Why use the `index` here for the label, rather than the name of the blog somehow? If a user has five blogs, are they expected to know which particular blog they're reblogging this on by its number somehow?

Reply:
Thanks for the tips, I have seen that before but didn't have the time to research last night.

And yes by index is not ideal. Currently my blog is only text, with no limit on characters. So I was scared of horribly long urls and wanted something neat just to implement behaviour.